### PR TITLE
Close out openjpeg25 migration

### PR DESCRIPTION
**Pull request by hmaarrfk**
This lets the recipe manage its own pinnings which are looser than the original ones.

https://github.com/conda-forge/openjpeg-feedstock/blob/main/recipe/meta.yaml#L16

@isuruf  is this kind of syntax allowed?

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -683,7 +681,7 @@ openexr:
 openh264:
   - '2.3'
 openjpeg:
-  - '2.4'
+  - >=2.5,<3
```
Comment (hmaarrfk) [inline recipe/conda_build_config.yaml:684]:
```suggestion
  - '>=2.5,<3'
```

**Review by ocefpaf — APPROVED** [review]

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -683,7 +681,7 @@ openexr:
 openh264:
   - '2.3'
 openjpeg:
-  - '2.4'
+  - '>=2.5,<3'
```
Comment (hmaarrfk) [inline recipe/conda_build_config.yaml:684]: For clarity, this is the new syntax I'm worried about breaking things

Comment (isuruf) [reply recipe/conda_build_config.yaml:684]: I'm not sure this wouldn't break stuff. Let's keep this at `2.5` for now and change to 2 in a few months.

Comment (isuruf) [reply recipe/conda_build_config.yaml:684]: Btw, I've changed this to 2 in https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/3410/files

Comment (hmaarrfk) [reply recipe/conda_build_config.yaml:684]: openjpeg is a "well behaved" packaged, likely, but it is possible that old builds are just "bad", with missing "pinnings" and do not have "run_exports".

I would prefer it if we could provide a lower bound, especially for things that are pinned to `'x`` only.

**Pull request closed without merging**